src/views/VenueDetails/BookVenue/index.jsx

```diff
@@ -1,12 +1,102 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import styles from './styles.module.css'
+import { Calendar } from './Calendar'
+import CircularProgress from '@mui/material/CircularProgress'
+import PhoneIcon from '@mui/icons-material/Phone'
+import MailIcon from '@mui/icons-material/Mail'
+import { FaFacebookF } from 'react-icons/fa'
+import { linkPaths } from '../../../LinksPaths'
+import { AiFillInstagram, AiOutlineTwitter } from 'react-icons/ai'
+import { GrLanguage } from 'react-icons/gr'
+
+const dayInMilliseconds = 24 * 60 * 60 * 1000
+let firstDayTime = null
+let secondDayTime = null
+let amountOfBookedDays = null
+
+const socialMediaLinks = [
+  { icon: <FaFacebookF size={'2em'} />, path: linkPaths.facebook },
+  { icon: <AiFillInstagram size={'3em'}/>, path: linkPaths.instagram },
+  { icon: <AiOutlineTwitter size={'3em'}/>, path: linkPaths.twitter },
+  { icon: <GrLanguage size={'2.5em'}/>, path: linkPaths.website }
+]
+
+export function BookVenue ({ venueDetails }) {
+  const [value, setValue] = useState([null, null])
+  const [costPerDay, setCostPerDay] = useState(null)
+
+  if (value[0]) {
+    const firstDayDateRawData = Object.values(value[0])
+    const firstDayDate = new Date(firstDayDateRawData[4], firstDayDateRawData[5], firstDayDateRawData[6])
+    firstDayTime = firstDayDate.getTime()
+    if (value[1]) {
+      const secondDayDateRawData = Object.values(value[1])
+      const secondDayDate = new Date(secondDayDateRawData[4], secondDayDateRawData[5], secondDayDateRawData[6])
+      secondDayTime = secondDayDate.getTime()
+    }
+    if (!value[1]) {
+      secondDayTime = firstDayTime
+    }
+    amountOfBookedDays = Math.round((Math.abs((firstDayTime - secondDayTime) / (dayInMilliseconds))) + 1)
+  }
+
+  useEffect(() => {
+    fetch('https://api.apilayer.com/exchangerates_data/convert?to=PLN&from=EUR&amount=1',
+      {
+        headers: {
+          apikey: 'DtbpehV28fDVm4s3sLtJl3vWCAcH8cZU'
+        }
+      })
+      .then((res) => res.json())
+      .then(currencyExchange => {
+        setCostPerDay(parseFloat(venueDetails.pricePerNightInEur * currencyExchange.result).toFixed(2))
+      })
+  }, [])
 
-export function BookVenue () {
   return (
```

Comment:
This template seems to be too big, and too repeatable. Refactor it to make it more readable